nextmv/nextmv/cloud/application.py

```
@@ -3867,70 +3867,48 @@ def __input_set_for_scenario(self, scenario: Scenario, scenario_id: str) -> Inpu
-                f"Valid types are: {[InputFormat.CSV_ARCHIVE, InputFormat.MULTI_FILE]}",
-            )
-
-    def __extract_config_format(self, configuration: Union[RunConfiguration, dict[str, Any]]) -> Any:
-        """Extract format from configuration, handling both RunConfiguration objects and dicts."""
-        if isinstance(configuration, RunConfiguration):
-            return configuration.format
-
-        if isinstance(configuration, dict):
-            config_format = configuration.get("format")
-            if config_format is not None and isinstance(config_format, dict):
-                return Format.from_dict(config_format) if hasattr(Format, "from_dict") else config_format
-
-            return config_format
-
-        raise ValueError("Configuration must be a RunConfiguration object or a dict.")
-
-    def __extract_input_type(self, config_format: Any) -> Any:
-        """Extract input type from config format."""
-        if isinstance(config_format, dict):
-            format_input = config_format.get("format_input") or config_format.get("input")
-            if format_input is None:
-                raise ValueError(
-                    "If dir_path is provided, RunConfiguration.format.format_input must also be provided.",
-                )
+    def __get_input_type(self, config: Union[RunConfiguration, dict[str, Any]]) -> Optional[InputFormat]:
+        """
+        Auxiliary function to extract the input type from the run configuration.
+        """
 
-            if isinstance(format_input, dict):
-                return format_input.get("input_type") or format_input.get("type")
+        if config is None:
+            return None
 
-            return getattr(format_input, "input_type", None)
+        if isinstance(config, dict):
+            try:
+                config = RunConfiguration.from_dict(config)
+            except Exception:
+                return None
```

This shouldn't be necessary, I think. If anything I would add a step somewhere that:
- Checks if it is an instance of `RunConfiguration`
- Checks if it is an instance of `dict`
- Raise a `ValueError` exception saying that  `RunConfiguration` is not a valid type.

---

Well, maybe what I said above is a first step, then I guess the second step, in case it is a `RunConfiguration`, is what you have here. However, instead of returning `None` I would raise a `ValueError` exception that clearly states that we cannot construct the `RunConfiguration` from the provided `dict`. This means that the dict is wrong.

---

You mean that it should error instead of ignore when parsing the dict fails. That is probably fair. I'll change it. :blush: